pytorch_lightning/utilities/seed.py

```diff
@@ -47,9 +47,11 @@ def seed_everything(seed: Optional[int] = None, workers: bool = False) -> int:
     min_seed_value = np.iinfo(np.uint32).min
 
     try:
+        # Mypy typing is ignored below as the code simplicity is prefered to mypy correctness. Also, possible errors
+        # are handled by the exception.
         if seed is None:
-            seed = os.environ.get("PL_GLOBAL_SEED")
-        seed = int(seed)
+            seed = os.environ.get("PL_GLOBAL_SEED")  # type: ignore
+        seed = int(seed)  # type: ignore
```

Comment:
I personally think we could refactor it, the following should match the functionality and comply with mypy

```python
    if seed is None:
        env_seed = os.environ.get("PL_GLOBAL_SEED")
        if env_seed is None:
            seed = _select_seed_randomly(min_seed_value, max_seed_value)
            rank_zero_warn(f"No seed found, seed set to {seed}")
        else:
            try:
                seed = int(env_seed)
            except ValueError:
                seed = _select_seed_randomly(min_seed_value, max_seed_value)
                rank_zero_warn(f"Invalid seed found: {repr(env_seed)}, seed set to {seed}")
    if not isinstance(seed, int):
        seed = int(seed)
```

It limits the scope of the try-except to only the parsing of what's in the environment variable, which is unavoidable as we don't have control over what's written there.

Note that calling `seed_everything("invalid")` would now fail whereas it didn't in the previous version, but I think that case should fail as it violates the typing annotation.

Thoughts @awaelchli ?

Reply:
I don't get it fully but make sure that the behavior does not change.

Reply:
`seed_everything("invalid")` would change but do you think it should be supported? Passing a non-number string would get you a different random seed each time, so I don't think it should

Reply:
yes that's probably safer.